cpp/include/raft/sparse/matrix/preprocessing.cuh

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <raft/core/device_csr_matrix.hpp>
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resource/cuda_stream.hpp>
+#include <raft/core/resources.hpp>
+#include <raft/sparse/matrix/detail/preprocessing.cuh>
+
+#include <optional>
+
+namespace raft::sparse::matrix {
+
+template <typename T1, typename T2, typename IdxT>
+void encode_bm25(raft::resources& handle,
+                 raft::device_vector_view<T1, IdxT> rows,
+                 raft::device_vector_view<T1, IdxT> columns,
```

Comment:
This is a COO format and I'd prefer to see it accept our raft::sparse::coo_matrix_view instead of accepting these arrays individually. You'll notice this is not widespread yet because we just created these new formats but new sparse functions should use them.

Reply:
Since we support the sparse formats below, we should remove these ones now that accept the raw arrays.